flink-python/src/main/java/org/apache/flink/python/PythonFunctionRunner.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.python;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * The base interface of runner which is responsible for the execution of Python functions.
+ *
+ * @param <IN> Type of the input elements.
+ */
+@Internal
+public interface PythonFunctionRunner<IN> {
+
+	/**
+	 * Prepares the Python function runner, such as preparing the Python execution environment, etc.
+	 */
+	void open() throws Exception;
```

Comment:
Can we add these methods later? I am asking because these methods have been implemented in its child class however have not been covered in any tests. 
Or can we add some tests to cover these methods?

Reply:
These methods can only be tested in integration test. What about revisiting these methods when adding the integration tests in FLINK-14018. However, I'm in favor of keeping them as they are part of the functionality of PythonFunctionRunner and are also very important part. Just removing them because they are not test right now seems wired for me.